plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/IngestModule.java

```diff
@@ -53,23 +55,23 @@ protected void configure() {
         binder().bind(PipelineStore.class).asEagerSingleton();
         binder().bind(SimulateExecutionService.class).asEagerSingleton();
 
-        addProcessor(GeoIpProcessor.TYPE, new GeoIpProcessor.Factory());
-        addProcessor(GrokProcessor.TYPE, new GrokProcessor.Factory());
-        addProcessor(DateProcessor.TYPE, new DateProcessor.Factory());
-        addProcessor(SetProcessor.TYPE, new SetProcessor.Factory());
-        addProcessor(RenameProcessor.TYPE, new RenameProcessor.Factory());
-        addProcessor(RemoveProcessor.TYPE, new RemoveProcessor.Factory());
-        addProcessor(SplitProcessor.TYPE, new SplitProcessor.Factory());
-        addProcessor(JoinProcessor.TYPE, new JoinProcessor.Factory());
-        addProcessor(UppercaseProcessor.TYPE, new UppercaseProcessor.Factory());
-        addProcessor(LowercaseProcessor.TYPE, new LowercaseProcessor.Factory());
-        addProcessor(TrimProcessor.TYPE, new TrimProcessor.Factory());
-        addProcessor(ConvertProcessor.TYPE, new ConvertProcessor.Factory());
-        addProcessor(GsubProcessor.TYPE, new GsubProcessor.Factory());
-        addProcessor(MetaDataProcessor.TYPE, new MetaDataProcessor.Factory());
+        addProcessor(GeoIpProcessor.TYPE, environment -> new GeoIpProcessor.Factory(environment.configFile()));
```

Comment:
>  why don't we just either wire the concrete factories into a map binder

This would be the preferred way, but because the ingest processor code needs to be ES agnostic I'm unable to add `Environment` class as constructor parameter. So I needed some kind of provider code that creates factory instances without adding Guice or ES related dependencies to the ingest framework. That is what I did here. So it kind of moves the pollution from the ingest framework side to the ingest plugin side. 

>  see if we can pass in the Env. directly to the Module ctor (just like it can accept Settings)

So you mean create a new Environment instance based on the provided setting in the module's constructor? That idea did cross my mind, but I only saw this way of providing `Environment` being used in tests. Then I thought maybe we could end up with inconsistencies (maybe no how things work now, but down the road) because we're not using the same Environment  instance that the `Node` class instance is using. But if we are okay with not using the same `Environment` instance for constructing the processor factories then I can remove the processor factory provider and instantiate factory instances directly in the `IngestModule` class.